overlord/snapstate/backend.go

```diff
@@ -55,7 +55,7 @@ type managerBackend interface {
 	CopySnapData(newSnap, oldSnap *snap.Info, meter progress.Meter) error
 	LinkSnap(info *snap.Info) error
 	StartServices(svcs []*snap.AppInfo, meter progress.Meter) error
-	StopServices(svcs []*snap.AppInfo, meter progress.Meter) error
+	StopServices(svcs []*snap.AppInfo, reason string, meter progress.Meter) error
```

Comment:
Stop reason should be a distinct type with some constants, to avoid and deter typos.

Reply:
Yeah, when I wrote this I was wondering this too. I dislike the extra verbosity but I agree about the usefulness of the static checking.